### PR TITLE
Claim restake redesign

### PR DESCRIPTION
**Pull request by nicomac**
Implements #298

TODO:
- Split UI into smaller components
- Verify if Claim and Restake calculations are correct _(I'm a bit unsure of this one)_

### CODE REVIEW

**Comment by emersoncloud** [comment]
`src/modules/components/Modal/ClaimAndRestake/SuccessfulClaim.tsx` is used by `ClaimAndRestake`, `StakeGGP`, `UnstakeGGP`. 

What data do we want to show in each of those cases when a claim is successful? 

What I think would be useful:
1. ClaimAndRestake: Amount Claimed, Amount Restaked, Collat Ratio
2. StakeGGP: Stake Amount, Total Staked, Collat Ratio
3. UnstakeGGP: Unstake Amount, Total Staked, Collat Ratio

We are showing `GGP Balance` right now in all cases but that feels different / unrelated to what these things are doing. And the user can check that in Metamask. Maybe we show it for ClaimAndRestake but I'm not sure. 

Here's how it looks right now. 

1. ClaimAndRestake
image

4. StakeGGP
image

5. UnstakeGGP
image

File: src/modules/components/Modal/ClaimAndRestake/SuccessfulClaim.tsx [hunk 1]
```diff
@@ -49,22 +53,49 @@ export const SuccessfulClaim: FunctionComponent<SuccessfulClaimProps> = ({
...
-            {Math.round(ggpStake).toLocaleString(undefined, {
-              maximumFractionDigits: 2,
-              minimumFractionDigits: 2,
-            })}
-          </span>
-        </Text>
+        {staked ? (
+          <Text
+            casing="uppercase"
+            className="flex justify-between border-b border-dashed border-gray-400 pb-2"
```
Comment (emersoncloud) [inline src/modules/components/Modal/ClaimAndRestake/SuccessfulClaim.tsx:59]: So much repetition in these two cases and collateralization

File: src/modules/components/Modal/ClaimAndRestakeModal.tsx [hunk 2]
```diff
@@ -1,117 +1,186 @@
...
+              {restakeAmount === 0 ? 'Claim All...' : 'Claim and Restake...'}
+            </button>
+          </Flex>
         </div>
-      </Modal>
-    </>
+      )}
+
+      {/* STEP TWO */}
+      {currentStep === 2 && !(isSuccess && !isLoading && claimData?.hash) && (
```
Comment (emersoncloud) [inline src/modules/components/Modal/ClaimAndRestakeModal.tsx:126]: This feels like it should be it's own component.

File: src/modules/components/Modal/ClaimAndRestakeModal.tsx [hunk 3]
```diff
@@ -1,117 +1,186 @@
...
+    setRestakeAmount(0)
+    setCurrentStep(1)
   }
 
   return (
-    <>
-      <Modal isOpen={isOpen} onClose={handleClose} {...modalProps}>
+    <Modal isOpen={isOpen} onClose={handleClose} {...modalProps}>
+      {/* STEP ONE */}
+      {currentStep === 1 && (
```
Comment (emersoncloud) [inline src/modules/components/Modal/ClaimAndRestakeModal.tsx:67]: Should this move into a separate component?

Comment (jacktrusler) [reply src/modules/components/Modal/ClaimAndRestakeModal.tsx:62]: Yea, I think so

**Comment by emersoncloud** [comment]
LGTM

**Pull request merged**